sentry-core/src/main/java/io/sentry/core/Scope.java

```diff
@@ -402,9 +413,30 @@ SessionPair startSession() {
       }
       previousSession = session;
 
-      session =
-          new Session(
-              options.getDistinctId(), user, options.getEnvironment(), options.getRelease());
+      String distinctId = null;
+
+      final User finalUser = getUser();
+      //      synchronized (userLock) {
+      if (finalUser != null) {
+        if (finalUser.getId() != null && !finalUser.getId().isEmpty()) {
+          distinctId = finalUser.getId();
+        } else if (finalUser.getEmail() != null && !finalUser.getEmail().isEmpty()) {
+          distinctId = finalUser.getEmail();
+        } else if (finalUser.getUsername() != null && !finalUser.getUsername().isEmpty()) {
+          distinctId = finalUser.getUsername();
+        }
+      }
+
+      // fallback to sentryDeviceId
+      if (distinctId == null) {
+        // if I add a transient field (not serializable) to the User object, I can get rid of the
+        // distinctId on options
+        // I cannot simply set on user.Id cus its gonna be a breaking change
```

Comment:
This comment made me wonder if we should just release 3.0.0 with sessions and by default report the installation id as both `distinctId` and `user.id`. With that fallback to secureId you added already. Would simplify things?

Reply:
if we do a major bump, I'd remove `secureAndroidId` once for all.
I'd set `installationId` by default onto `Scope.user.id` on SDK init if `cacheUserForSessions` is disabled and if enabled, the cached `userId`.

this would make `event.user.id` and `session.distinctId` normalized.

Reply:
That makes sense. Also because a btter fallback is to tell users to call secure.AndroidId and set to the user.id by themselves